server/game/cards/02_SHD/events/CalculatedLethality.ts

```diff
@@ -26,11 +26,15 @@ export default class CalculatedLethality extends EventCard {
             },
             then: (thenContext) => ({
                 title: 'For each upgrade that was on that unit, give an Experience token to a friendly unit.',
-                immediateEffect: AbilityHelper.immediateEffects.distributeExperienceAmong({
-                    amountToDistribute: thenContext.targets.upgradeAmount,
-                    cardTypeFilter: WildcardCardType.Unit,
-                    controller: RelativePlayer.Self,
-                    canChooseNoTargets: false,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
```

Comment:
Simpler to use `thenCondition` instead of wrapping the whole thing in conditional